### DISCUSSION TITLE
Exporting part of a Nickel file?

### DISCUSSION THREAD

**Discussion by r-k-b** [tweag/nickel#825] [Q&A] [answered]
I'd like have a single Nickel file from which I can export multiple related config files, such as a `docker-compose.yaml` plus an `nginx.conf`, plus an html file.
Is it possible to `export` a certain path from a Nickel file? Or is it necessary to have a Nickel file for each export format?

---

For example, Dhall allows exporting to both yaml and bash from a single file like so:

```
dhall-to-yaml <<< '(./company.dhall).bios'

dhall-to-bash --declare EMAILS <<< '(./company.dhall).emails'
```

**Comment by yannham** [comment, marked as answer]
It's slightly more verbose, but would this fit the bill?

```
nickel export --format yaml <<< '(import "company.ncl").bios'
nickel export --format raw <<< '(import "company.ncl").emails'
```

**Reply by yannham** [reply]
(I'm not really familiar with dhall-to-bash, so the second example may need more massaging. But the `raw` format basically evaluates a value, expects a string, and then print it out)

**Reply by r-k-b** [reply]
Excellent, thanks yannham!
